### DISCUSSION TITLE
How to migrate the design from Icicle kit to a new device.  Debuging is fail.

### DISCUSSION THREAD

**Discussion by xiyanni** [polarfire-soc/polarfire-soc-documentation#115] [HSS and U-Boot] [answered]
Customer is using MPFS025T, HSS version is 2021.11. 
They used the icicle kit before their own board is ready.  
After they programmed the envm, when they planned to debug HSS via SoftConsole, an error happened. "program file does not exist" 
How to migrate the design from Icicle kit to a new device? How many parts need to be update?

Below is the details:
/*************************************************************************************************************/
Info : only one transport option; autoselect 'jtag'
Info : Hardware thread awareness created
do_board_reset_init
Info : Listening on port 6666 for tcl connections
Info : Listening on port 4444 for telnet connections
Info : No Embedded FlashPro6 (revision B) devices found
Info : clock speed 6000 kHz
Info : JTAG tap: mpfs.cpu tap/device found: 0x0f8531cf (mfg: 0x0e7 (GateField), part: 0xf853, ver: 0x0)
Warn : JTAG tap: mpfs.cpu       UNEXPECTED: 0x0f8531cf (mfg: 0x0e7 (GateField), part: 0xf853, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 1 of 15: 0x0f8581cf (mfg: 0x0e7 (GateField), part: 0xf858, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 2 of 15: 0x0f8781cf (mfg: 0x0e7 (GateField), part: 0xf878, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 3 of 15: 0x0f8771cf (mfg: 0x0e7 (GateField), part: 0xf877, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 4 of 15: 0x0f8181cf (mfg: 0x0e7 (GateField), part: 0xf818, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 5 of 15: 0x0f8381cf (mfg: 0x0e7 (GateField), part: 0xf838, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 6 of 15: 0x0f83c1cf (mfg: 0x0e7 (GateField), part: 0xf83c, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 7 of 15: 0x0f8191cf (mfg: 0x0e7 (GateField), part: 0xf819, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 8 of 15: 0x0f8391cf (mfg: 0x0e7 (GateField), part: 0xf839, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 9 of 15: 0x0f83d1cf (mfg: 0x0e7 (GateField), part: 0xf83d, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 10 of 1: 0x0f81a1cf (mfg: 0x0e7 (GateField), part: 0xf81a, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 11 of 1: 0x0f83a1cf (mfg: 0x0e7 (GateField), part: 0xf83a, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 12 of 1: 0x0f83e1cf (mfg: 0x0e7 (GateField), part: 0xf83e, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 13 of 1: 0x0f81b1cf (mfg: 0x0e7 (GateField), part: 0xf81b, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 14 of 1: 0x0f83b1cf (mfg: 0x0e7 (GateField), part: 0xf83b, ver: 0x0)
Error: JTAG tap: mpfs.cpu expected 15 of 1: 0x0f83f1cf (mfg: 0x0e7 (GateField), part: 0xf83f, ver: 0x0)
Error: Trying to use configured scan chain anyway...
Warn : Bypassing JTAG setup events due to errors
Info : datacount=2 progbufsize=16
Info : Disabling abstract command reads from CSRs.
Info : Core 0 could not be made part of halt group 1.
Info : Examined RISC-V core; found 5 harts
/*************************************************************************************************************/

**Comment by hughbreslin** [comment, marked as answer]
Hi @xiyanni 

for:
> program file does not exist

Does the program file specified / the path to the program file match the debug configuration in SoftConsole? For example in the screenshot below the debug configuration points to `Default/hss-l2lim.elf` and the file exists in the project explorer on the left in the `Default` folder:
![image](https://user-images.githubusercontent.com/44862881/177277010-2b61f799-03d5-45f0-9f32-f36e10e8e668.png)

Once they've built the HSS with the XML for the new design, configured the build for the services that are needed and managed to program the eNVM they should be doing pretty well in terms of migrating.

**Comment by xiyanni** [comment]
The program file specified / the path to the program file matches the debug configuration in SoftConsole. They have degbug the HSS on Icicle kit successfully. This link is helpful for customer. https://github.com/orgs/polarfire-soc/discussions/113

But now they are stuck at the debugging of HSS via Softconsole.

**Reply by hughbreslin** [reply]
Ok so the debug issue has only come up with the custom board? Has any debug session been able to connect for embedded software, e.g one of the test projects for the mpfs-hal? Have they seen the 

> Error: JTAG tap: mpfs.cpu expected 1 of 15: 0x0f8581cf (mfg: 0x0e7 (GateField), part: 0xf858, ver: 0x0)

prints before or is this new?

**Reply by xiyanni** [reply]
This error is new on customer board which is using MPFS025T. They did not see the error on the demo board.
